### DISCUSSION TITLE
does cypress create temp files? keep seeing this downloads.html.crdownload and downloads.html

### DISCUSSION THREAD

**Discussion by sandrodesouza** [cypress-io/cypress#28859] [Questions and Help]
does cypress create temp files? keep seeing this "downloads.html.crdownload" and "downloads.html" during the "cypress run" execution. Using chrome browser, probably related. These files are deleted after run finishes, it is not a issue to me. Just wondering if is just me, I didn't find anyone talking about this. Files are in binary, I cant see the content. Is placed in the folder cypress/downloads. 
Btw I don't have any command to download a file in my tests.

**Comment by RKoornneef** [comment]
@sandrodesouza I see the same thing happening. But in my case it is not deleted after test run. It seems to be created both on using cypress UI as well as when running headless (cypress -run from terminal).

<img width="294" alt="Screenshot 2024-02-28 at 7 11 04 PM" src="https://github.com/cypress-io/cypress/assets/116634726/9e16ca8a-1a59-4626-8c41-45a0fb17e484">

**Comment by scottohara** [comment]
Anecdotally, I tend to see this (a `downloads.html` file with what appears to be binary content) on the first run after updating to a new version of Cypress.

That is, when you see the message "It looks like this is your first time using Cypress: <version>" in the console and it downloads/verifies the Cypress binary.

My guess is that this first run check is somehow related to the `downloads.html` file appearing in the downloads folder.

**Reply by scottohara** [reply]
Further investigation suggests that the mysterious `downloads.html` file may in fact be somehow related to Chrome "Optimization Guide Prediction" models; and nothing really to do with Cypress itself.

What I noticed is that when I launch an e2e session in Chrome (`cypress open --e2e --browser chrome`), on occasion I will see the following log entries in the terminal (stdout) for the Cypress process:

```
GET /downloads?name=1711378964&target=OPTIMIZATION_TARGET_CLIENT_SIDE_PHISHING 200 2356.424 ms - -
GET /downloads?name=1673999601&target=OPTIMIZATION_TARGET_PAGE_VISIBILITY 200 2244.424 ms - -
GET /downloads?name=1691042511&target=OPTIMIZATION_TARGET_NEW_TAB_PAGE_HISTORY_CLUSTERS_MODULE_RANKING 200 2129.876 ms - -
GET /downloads?name=1689043206&target=OPTIMIZATION_TARGET_VISUAL_SEARCH_CLASSIFICATION 200 2212.676 ms - -
```

Briefly these log entries coincide with a `downloads.html.crdownload` file in the downloads folder; which is then cleaned up moments later.

A quick google for those target strings (`OPTIMIZATION_TARGET_*`) finds references to them in the Chromium source code repository.

Furthermore, I found a Cypress issue (https://github.com/cypress-io/cypress/issues/22622) from a couple of years ago, where it was asked whether these optimisation guide models should be disabled; and it seems that it was decided not to disable.

In which case, I think the next question should be:  how can we ensure that any artifacts from these "optimization guide" downloads are not left lying around in the downloads folder?

The issue I have with this file is that we don't have `cypress/downloads` in our .gitignore file (because normally that directory doesn't exist as we don't use any commands that download anything), so it is possible that this spurious file may get accidentally committed if we're not careful.  Yes, I could simply .gitignore the downloads directory to avoid this, but I shouldn't need to ignore a directory that otherwise isn't supposed to be present when unused.

**Reply by distante** [reply]
ah! I just got it for the first time. Still an "issue" in 2025

**Comment by jennifer-shehane** [comment]
These Optimization guide/hint features are now disabled by default in Cypress 15.6.0
